lib/Model/Importer/Binary/PECOFFImporter.cpp

```diff
@@ -74,22 +74,48 @@ Error PECOFFImporter::import() {
     }
     const object::coff_section *CoffRef = *SecOrErr;
 
-    // VirtualSize might be larger than SizeOfRawData (extra data at the end of
-    // the section) or viceversa (data mapped in memory but not present in
-    // memory, e.g., .bss)
-    uint64_t SegmentSize = std::min(CoffRef->VirtualSize,
-                                    CoffRef->SizeOfRawData);
-
     MetaAddress Start = ImageBase + u64(CoffRef->VirtualAddress);
     Segment Segment({ Start, u64(CoffRef->VirtualSize) });
 
     Segment.StartOffset = CoffRef->PointerToRawData;
-    Segment.FileSize = Segment.StartOffset + SegmentSize;
+
+    // VirtualSize might be larger than SizeOfRawData (extra data at the end of
+    // the section) or viceversa (data mapped in memory but not present in
+    // memory, e.g., .bss)
+    Segment.FileSize = CoffRef->SizeOfRawData;
+
+    // Since it is possible that the file size if greater than VirtualSize
+    // because SizeOfRawData is rounded, but VirtSize is not,
+    // we work it around here.
+    // TODO: Revisit this.
+    if (Segment.FileSize > Segment.VirtualSize)
+      Segment.VirtualSize = Segment.FileSize;
 
     Segment.IsReadable = CoffRef->Characteristics & COFF::IMAGE_SCN_MEM_READ;
     Segment.IsWriteable = CoffRef->Characteristics & COFF::IMAGE_SCN_MEM_WRITE;
     Segment.IsExecutable = CoffRef->Characteristics
                            & COFF::IMAGE_SCN_MEM_EXECUTE;
+
+    // NOTE: Unlike ELF, PE/COFF does not have segments. Instead,
+    // it has sections only. All the raw data in a section must be loaded
+    // contiguously. Segments just map file range to virtual address space
+    // range.
```

Comment:
`it` should be on the previous line. Sorry to pick on you on this small things, but I'd like to make sure each developer has an automated way to handle comment wrapping (depending on his own workflow) so that this never happens.

Reply:
Oh...submlime :/ I thought I've set it up...
However, please don't be sorry, I have no problem with the comments like that, I am sorry it happened, I pray the sublime setting works now lol
It is sad the clang-format cannot detect and format this (or it can?!)...

Reply:
`clang-format` shortens long lines but it doesn't do the other way round (since the existing short lines might be there for a reason). So, making the comment as long as you want and then running `clang-format` should do the trick.